### PR TITLE
Feature/issue 552 - New vehicle data - FuelRange

### LINKED ISSUE #552: [SDL 0072] New vehicle data - FuelRange
Proposal: [New vehicle data - FuelRange](https://github.com/smartdevicelink/sdl_evolution/blob/master/proposals/0072-New-vehicle-data-FuelRange.md)

> This proposal is about adding a new vehicle data to get the distance a vehicle can travel with the current level of fuel.

Review: https://github.com/smartdevicelink/sdl_evolution/issues/207

Steering Committee Decision:

> The Steering Committee has voted to accept this proposal with revisions. It was agreed that FuelType and range be optional and not mandatory, based on the reasons provided in the comments on the issue.  These revisions have now been made to the referenced proposal .md file.

### PR DESCRIPTION
**Pull request by mrapitis**
Fixes #552 

This PR is **ready** for review.

## Risk

This PR makes **minor** API changes.

## Testing Plan

Added new unit tests and expanded existing ones

## Summary

Added Vehicle Data RPC's, structs, enums as defined in the fuel range proposal.

## CLA

- [x] I have signed the CLA

### CODE REVIEW

**Comment by mrapitis** [comment]
@brettywhite @bilal-alsharifi please review when time permits. thanks!

File: sdl_android/src/androidTest/java/com/smartdevicelink/test/rpc/requests/UnsubscribeVehicleDataTests.java [hunk 1]
```diff
@@ -91,7 +92,8 @@ protected JSONObject getExpectedParameters(int sdlVersion) {
             result.put(UnsubscribeVehicleData.KEY_AIRBAG_STATUS, Test.GENERAL_BOOLEAN);
             result.put(UnsubscribeVehicleData.KEY_EMERGENCY_EVENT, Test.GENERAL_BOOLEAN);
             result.put(UnsubscribeVehicleData.KEY_CLUSTER_MODE_STATUS, Test.GENERAL_BOOLEAN);
-            result.put(UnsubscribeVehicleData.KEY_MY_KEY, Test.GENERAL_BOOLEAN);            
+            result.put(UnsubscribeVehicleData.KEY_MY_KEY, Test.GENERAL_BOOLEAN);
+			result.put(UnsubscribeVehicleData.KEY_FUEL_RANGE, Test.GENERAL_BOOLEAN);
```
Comment (bilal-alsharifi) [inline sdl_android/src/androidTest/java/com/smartdevicelink/test/rpc/requests/UnsubscribeVehicleDataTests.java]: It looks like there are extra spaces/tabs here and in some other places in the modifications

**Comment by mrapitis** [comment]
@bilal-alsharifi please re-review when time permits. thanks!

**Comment by bilal-alsharifi** [comment]
Thanks, @mrapitis. There are still some minor spacing issues in the following files:

OnVehicleDataTests.java
GetVehicleDataTests.java
SubscribeVehicleDataTests.java
GetVehicleDataResponseTests.java
SubscribeVehicleDataResponseTest.java
GetVehicleData.java
GetVehicleDataResponse.java
OnVehicleData.java
SubscribeVehicleDataResponse.java
UnsubscribeVehicleDataResponse.java

**Comment by mrapitis** [comment]
@bilal-alsharifi thanks for the review, please check once more when available.

**Comment by bilal-alsharifi** [comment]
@mrapitis everything looks good to me now. Thanks.

**Pull request merged**